app/services/claims/state_transition_message_builder.rb

```diff
@@ -0,0 +1,36 @@
+module Claims
+  class StateTransitionMessageBuilder
+    attr_reader :reasons, :reason_text
+
+    def initialize(state, reasons, reason_text)
+      @state = state
+      @reasons = reasons
+      @reason_text = reason_text
+    end
+
+    def call
+      @reasons_message ||= reasons_message
+    end
+
+    private
+
+    def reason_details(reason, reason_text)
+      ClaimStateTransitionReason.get(reason, reason_text)
+    end
+
+    def long_reasons
+      reasons.each_with_object([]) do |reason, arr|
+        details = reason_details(reason, reason_text)
+        arr << "#{details.description}: #{details.long_description}"
+      end
+    end
+
+    def reasons_message
+      <<~MSG
+        Your claim has been #{@state}:
```

Comment:
💭 Consider defining this copy in the locales with both _state_ and _long_reasons_ as variables that can be interpolated.

Reply:
if i had time i had considered I18n'ing all the hash strings. seems more consistent to have one approach.